plugins/dashboard/routes.go

```diff
@@ -139,176 +188,53 @@ func enforceMaxOneDotPerURL(next echo.HandlerFunc) echo.HandlerFunc {
 	}
 }
 
-func setupRoutes(e *echo.Echo) {
-
-	e.Pre(enforceMaxOneDotPerURL)
-
-	e.GET("/ws", websocketRoute)
-	e.GET("/", indexRoute)
-
-	e.GET("/static/*", passThroughRoute)
-	e.GET("/branding/*", passThroughRoute)
-	e.GET("/favicon/*", passThroughRoute)
-
-	// Hot reload code
-	e.GET("/main*.js", passThroughRoute)
-
-	// Pass all the explorer request through to the local rest API
-	e.GET("/api/*", passThroughAPIRoute)
-	e.GET("/api/plugins/spammer*", passThroughSpammerRoute)
-
-	// Everything else fallback to index for routing.
-	e.GET("*", indexRoute)
-
-	e.HTTPErrorHandler = func(err error, c echo.Context) {
-		c.Logger().Error(err)
-
-		var statusCode int
-		var message string
-
-		switch errors.Cause(err) {
+func authRoute(c echo.Context) error {
 
-		case echo.ErrNotFound:
-			c.Redirect(http.StatusSeeOther, "/")
-			return
-
-		case echo.ErrUnauthorized:
-			statusCode = http.StatusUnauthorized
-			message = "unauthorized"
-
-		case ErrForbidden:
-			statusCode = http.StatusForbidden
-			message = "access forbidden"
-
-		case ErrInternalError:
-			statusCode = http.StatusInternalServerError
-			message = "internal server error"
-
-		case ErrNotFound:
-			statusCode = http.StatusNotFound
-			message = "not found"
-
-		case ErrInvalidParameter:
-			statusCode = http.StatusBadRequest
-			message = "bad request"
+	type loginRequest struct {
+		JWT      string `json:"jwt"`
+		User     string `json:"user"`
+		Password string `json:"password"`
+	}
 
-		default:
-			statusCode = http.StatusInternalServerError
-			message = "internal server error"
-		}
+	request := &loginRequest{}
 
-		message = fmt.Sprintf("%s, error: %+v", message, err)
-		c.String(statusCode, message)
+	if err := c.Bind(request); err != nil {
+		return errors.WithMessagef(ErrInvalidParameter, "invalid request, error: %s", err)
 	}
-}
-
-func websocketRoute(ctx echo.Context) error {
-	defer func() {
-		if r := recover(); r != nil {
-			log.Errorf("recovered from panic within WS handle func: %s", r)
-		}
-	}()
 
-	// this function sends the initial values for some topics
-	sendInitValue := func(client *websockethub.Client, initValuesSent map[byte]struct{}, topic byte) {
-		if _, sent := initValuesSent[topic]; sent {
-			return
+	if len(request.JWT) > 0 {
+		// Verify JWT is still valid
+		if !jwtAuth.VerifyJWT(request.JWT) {
```

Comment:
Can we do an early return if the auth is still correct, or do we need to issue a new one?

```go
	return c.JSON(http.StatusOK, map[string]string{
		"jwt": request.JWT,
	})``